Objects/exceptions.c

```diff
@@ -3015,7 +3100,7 @@ UnicodeEncodeError_init(PyObject *self, PyObject *args, PyObject *kwds)
         return -1;
     }
 
-    PyUnicodeErrorObject *exc = (PyUnicodeErrorObject *)self;
+    PyUnicodeErrorObject *exc = PyUnicodeError_CAST(self);
```

Comment:
Why switch these? `CAST` macros are only useful for header files, because they prevent C++ compilation from failing.

Reply:
Because there is an assert in this cast.

Reply:
> CAST macros are only useful for header files,

There are also useful for semantics when dealing with UBSan (again, we used this logic when we were fixing the other cases of UBSan).

Reply:
> There are also useful for semantics when dealing with UBSan

Oh! What does it help with?

Reply:
I meant, we use those macros to convert self -> typed self. It's a common pattern in the code. Now, at runtime, the fact that there is an assert *is* good because we don't want to call the function with a generic PyObject * (in other words, we restrict the signature at runtime because there is no templating in C).

Reply:
Ok, I'll take your word for it.